generated/src/aws-cpp-sdk-dynamodb/source/DynamoDBClientConfiguration.cpp

```diff
@@ -0,0 +1,70 @@
+﻿/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/dynamodb/DynamoDBClientConfiguration.h>
+
+namespace Aws
+{
+namespace DynamoDB
+{
+
+
+bool IsEndpointDiscoveryEnabled(const Aws::String& endpointOverride, const Aws::String &profileName)
+{
+  bool enabled = false;
+  if (!endpointOverride.empty())
+  {
+    enabled = false;
+  }
+  else
+  {
+    static const char* AWS_ENABLE_ENDPOINT_DISCOVERY_ENV_KEY = "AWS_ENABLE_ENDPOINT_DISCOVERY";
+    static const char* AWS_ENABLE_ENDPOINT_DISCOVERY_PROFILE_KEY = "AWS_ENABLE_ENDPOINT_DISCOVERY";
+    static const char* AWS_ENABLE_ENDPOINT_ENABLED = "true";
+    static const char* AWS_ENABLE_ENDPOINT_DISABLED = "false";
+
+    Aws::String enableEndpointDiscovery = Client::ClientConfiguration::LoadConfigFromEnvOrProfile(
```

Comment:
The profile file can change and we must reload it.
The env variable technically won't change after the app launch, but it can be changed during the runtime by the application itself.
So it works as it worked before, and it is client-dependent.